kovri-util: add logs and data-dir options

MoroccanMalinois opened the pull request:
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.

*Place an X inside the bracket (or click the box in preview) to confirm*
- [X] I confirm.

---
su3file need the certificates in data-dir and log options are necessary to be able "to pipe" the output.  

Also Referencing #563

anonimal commented:
Please rebase.

MoroccanMalinois commented:
Rebased

guzzijones commented on src/util/base.cc:
```diff
@@ -104,18 +104,15 @@ bool process(
 
 bool BaseCommand::Impl(
     const std::string& cmd_name,
-    int argc,
```
Why the move away from argc and argv?  I now cannot parse out -a as an option for benchmark .  It thinks it is -all for kovri-util, when i run kovri-util benchark -a the help prints out for me and the -all is NOT run anyway.

guzzijones replied:
I think the correct solution was to use all options AFTER the positional argument as options for the benchmark test. The rerun the bpo::command_line_parser. As it currently stands i wont be able to use any of the options for main --all, -a, etc.  I would rather put it back before I push.

MoroccanMalinois commented:
I replaced (argc, argv) with a vector. In fact, i changed the logic in main because otherwise the arguments managed in main had to be put first, which is not how most cli works. I think it's better now. But the effect is that we can't use the same flag (-a) in the main and in a sub command. I suggest you remove the option -a in the main, since it was not a terrible idea to begin with :) 

Sorry for the inconvenience.

guzzijones commented:
' i changed the logic in main because otherwise the arguments managed in main had to be put first,'  I actually thought that was great.  I am going to wait until anonimal chimes in on this.  I am waiting for a fix for the log level anyway.

MoroccanMalinois commented:
The fix for the log-level is simply changing 1 by 3 in https://github.com/monero-project/kovri/blob/master/src/util/main.cc#L70
you can also use "./build/kovri-util --log-level 3 ...."
